network/framework/src/peer/mod.rs

```diff
@@ -270,12 +270,29 @@ where
                 // Drive the queue of pending inbound rpcs. When one is fulfilled
                 // by an upstream protocol, send the response to the remote peer.
                 maybe_response = self.inbound_rpcs.next_completed_response() => {
-                    if let Err(err) = self.inbound_rpcs.send_outbound_response(&mut write_reqs_tx, maybe_response).await {
-                        warn!(
-                            NetworkSchema::new(&self.network_context).connection_metadata(&self.connection_metadata),
-                            error = %err,
-                            "{} Error in handling inbound rpc request, error: {}", self.network_context, err,
-                        );
+                    if let Err(error) = self.inbound_rpcs.send_outbound_response(&mut write_reqs_tx, maybe_response).await {
+                        // It's quite common for applications to drop an RPC request.
+                        // If this happens, we want to avoid logging a warning/error
+                        // (as it makes the logs noisy). Otherwise, we log normally.
+                        let network_schema = NetworkSchema::new(&self.network_context)
+                            .connection_metadata(&self.connection_metadata);
+                        let error_string = format!("{} Error in handling inbound rpc request, error: {}", self.network_context, error);
+                        match error {
+                            RpcError::UnexpectedResponseChannelCancel => {
```

Comment:
The problem is not just log level, but more about not providing enough useful information.. I'd like to at least know which rpc method is this error about, or I think even info! is too high..

Reply:
FYI: I've made this `debug!` now. There's not a lot of useful information we can display without a much bigger refactor, so it makes sense to ignore this log in most cases 😄